src/components/time-on-target-chart/TimeOnTargetChart.svelte

```diff
@@ -1,42 +1,51 @@
 <script lang="ts">
     import { THARGOID_TYPES } from '../../typescript/data/thargoidData'
     import InterceptorData from './interceptor-data/InterceptorData.svelte';
-    import { selectedDistributor, sdpsExtraDelay } from '../../typescript/store';
-    import { MIN_GAUSS_DELAY_MS, MS_PER_MINUTE } from '../../typescript/util'
+    import { selectedDistributor, sdpsExtraDelay, selectedWeapons } from '../../typescript/store';
+    import { MIN_GAUSS_DELAY_MS, MS_PER_MINUTE, MM_IN_MS } from '../../typescript/util'
 
     let shipName = '';
     $: delay = Math.round($sdpsExtraDelay) + MIN_GAUSS_DELAY_MS;
     $: bpm = Math.round(MS_PER_MINUTE / delay * 4);
+    $: delayInMm = (delay * MM_IN_MS).toFixed(3)
+    $: $selectedWeapons, hideDelay();
 
     let accuracy = '100';
     let ammo = "Basic";
+    let haveGaussSelected = false;
+    let toggleMm = false;
 
-    const distributorEngineering = () => {
+    const distributorEngineering = (): string => {
         const blueprint = $selectedDistributor.blueprint;
         const effect = $selectedDistributor.experimentEffect;
 
-        if ($selectedDistributor.class === 'G') {
+        if ($selectedDistributor.rating === 'G') {
             return 'Guardian';
         } else if (!blueprint) {
             return 'Not engineered';
         }
 
         return `${blueprint.toUpperCase()} + ${effect ? effect.toUpperCase() : 'No effect'}`;
-    }
+    };
+
+    const hideDelay = (): void => {
+        let numberOfGauss = 0;

```

Comment:
I'm pretty sure doing:
`haveGaussSelected = Object.values($selectedWeapons).filter(weapon => weapon.weaponName === 'gausscannon').length > 0;`

achieves the same goal.

Reply:
Hax, but fixed